### ISSUE TITLE
Trainable CRF loss-function

### ISSUE THREAD

**Issue by ipoletaev** [fchollet/keras#3638]
Hello! I want to implement the neural network model with CRF-based objective function via Keras.
So, in order to do it, the loss-function should store and update during the train a few matrixes. 
How can I do this? Moreover this layer should be able in test or predict mode to give another output - a most likely sequence of tags. As I've understood - there's no way to solve this issue without changes in the core code of Keras.

Thanks for help!

**Comment by alex-j-j** [comment]
I am also interested in something similar (specifically, the "sentence-level log-likelihood" loss from the 'NLP (almost) from scratch'-paper).
